e2etest/bgp_test.go

```diff
@@ -1151,8 +1152,82 @@ var _ = ginkgo.Describe("BGP", func() {
 		table.Entry("IPV6", "fc00:f853:0ccd:e799::/116", "ipv6", "fc00:f853:ccd:e800::1/128"),
 	)
 
+	ginkgo.Context("validate FRR running configuration", func() {
+		ginkgo.It("Full BFD profile", func() {
+			bfdProfile := config.BfdProfile{
+				Name:             "fullBFDProfile1",
+				ReceiveInterval:  uint32Ptr(93),
+				TransmitInterval: uint32Ptr(95),
+				EchoInterval:     uint32Ptr(97),
+				EchoMode:         boolPtr(true),
+				PassiveMode:      boolPtr(true),
+				MinimumTTL:       uint32Ptr(253),
+			}
+
+			configData := config.File{
+				Pools: []config.AddressPool{
+					{
+						Name:      "bgp-test",
+						Protocol:  config.BGP,
+						Addresses: []string{v4PoolAddresses},
+					},
+				},
+				Peers:       peersForContainers(frrContainers, "ipv4"),
+				BFDProfiles: []config.BfdProfile{bfdProfile},
+			}
+
+			configData.Peers = withBFD(configData.Peers, bfdProfile.Name)
+
+			for i := range configData.Peers {
+				configData.Peers[i].KeepaliveTime = "13s"
+				configData.Peers[i].HoldTime = "57s"
+			}
+
+			err := configUpdater.Update(configData)
+			framework.ExpectNoError(err)
+
+			speakerPods := getSpeakerPods(cs)
+			for _, pod := range speakerPods {
+				podExecutor := executor.ForPod(pod.Namespace, pod.Name, "frr")
+
+				Eventually(func() string {
+					// We need to assert against the output of the command as a bare string, as
+					// there is no json version of the command.
+					cfgStr, err := podExecutor.Exec("vtysh", "-c", "show running-config")
+					framework.ExpectNoError(err)
+					return cfgStr
+				}, 1*time.Minute).Should(
+					And(
+						ContainSubstring("log file /etc/frr/frr.log informational"),
+						WithTransform(substringCount("\n profile fullBFDProfile1"), Equal(1)),
+						ContainSubstring("receive-interval 93"),
+						ContainSubstring("transmit-interval 95"),
+						ContainSubstring("echo-interval 97"),
+						ContainSubstring("minimum-ttl 253"),
+						ContainSubstring("passive-mode"),
+						ContainSubstring("echo-mode"),
+
+						ContainSubstring("timers 13 57"),
+
+						// TODO - Current implementation lacks of selecting a deterministic routerID
+						// given a specific configuration.
+						//ContainSubstring("bgp router-id ???.???.???.???"),
+					),
+				)
+			}
+		})
+	})
+
 })
 
+// substringCount creates a Gomega transform function that
+// counts the number of occurrences in the subject string.
+func substringCount(substr string) interface{} {
```

Comment:
Why does this return interface{}?

Reply:
It's what `gomega.WithTransform(...)` expects to receive. I'm not sure a signature like:

```
func substringCount(substr string) func(action string) int {
    ...
```

would be more readable or it can be reused someway in other point.
Do you spot any good point on this?

Reply:
tbh not sure what's better (or even declaring it as an anonymous only where it's used). I guess it's fine if we keep it this way.